compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -40,11 +43,15 @@ struct ConstToPat<'a, 'tcx> {
     span: Span,
     param_env: ty::ParamEnv<'tcx>,
 
-    // This tracks if we signal some hard error for a given const value, so that
+    // This tracks if we saw some error or lint for a given const value, so that
     // we will not subsequently issue an irrelevant lint for the same const
     // value.
     saw_const_match_error: Cell<bool>,
```

Comment:
This is a bit sketchy in that a hard error might be gated on `saw_const_match_error` being `false`, but if there was only a lint, we're now emitting only lints and no hard errors.

I'd suggest having another `Cell<bool>` for lints.

Reply:
For now, this will cause some diagnostic duplication where we get a future incompat lint even though an error has been emitted, because the diagnostic is emitted before the error. This will go away when we remove the `NonStructuralMatchTy` fallback logic.

Reply:
nevermind, I got rid of that duplication, too